deploy/roles/elasticsearch/tasks/main.yml

```diff
@@ -45,15 +45,15 @@
 
 - name: Open ports for Elasticsearch transport communication
   command: "iptables -I {{ iptables_chain }} 3 -m state --state NEW -p tcp --dport {{ es_transport_ports }} -j ACCEPT"
-  when: iptables_config
+  when: iptables_config and proxy_server_deployed
 
 - name: Open ports for Elasticsearch http communication
   command: "iptables -I {{ iptables_chain }} 3 -m state --state NEW -p tcp --dport {{ es_http_ports }} -s {{ hostvars[groups['proxy_server'][0]]['ansible_ssh_host'] }} -j ACCEPT"
-  when: iptables_config
+  when: iptables_config and proxy_server_deployed
```

Comment:
@stirlingv Rather than creating a new variable, could you just say `iptables_config and 'proxy_server' in groups`?